core/assets/js/app.js

```diff
@@ -13,96 +13,232 @@ import "../css/app.css";
+import "alpine-magic-helpers/dist/component";
+import Alpine from "alpinejs";
+import "phoenix_html";
+import { Socket } from "phoenix";
+import { LiveSocket } from "phoenix_live_view";
 import { decode } from "blurhash";
-import {urlBase64ToUint8Array} from "./tools"
+import { urlBase64ToUint8Array } from "./tools";
 
 window.blurHash = () => {
-    return {
-        show: true,
-        showBlurHash() {
-            return this.show!== false;
-        },
-        hideBlurHash() {
-            this.show= false;
-        },
-        render() {
-            const canvas = this.$el.getElementsByTagName("canvas")[0];
-            if (canvas.dataset.rendered) { return }
-            const blurhash = canvas.dataset.blurhash
-            const width = parseInt(canvas.getAttribute("width"), 10)
-            const height = parseInt(canvas.getAttribute("height"), 10)
-            const pixels = decode(blurhash, width, height);
-            const ctx = canvas.getContext("2d");
-            const imageData = ctx.createImageData(width, height);
-            imageData.data.set(pixels);
-            ctx.putImageData(imageData, 0, 0);
-            canvas.dataset.rendered = true
-        }
-    }
-}
+  return {
+    show: true,
+    showBlurHash() {
+      return this.show !== false;
+    },
+    hideBlurHash() {
+      this.show = false;
+    },
+    render() {
+      const canvas = this.$el.getElementsByTagName("canvas")[0];
+      if (canvas.dataset.rendered) {
+        return;
+      }
+      const blurhash = canvas.dataset.blurhash;
+      const width = parseInt(canvas.getAttribute("width"), 10);
+      const height = parseInt(canvas.getAttribute("height"), 10);
+      const pixels = decode(blurhash, width, height);
+      const ctx = canvas.getContext("2d");
+      const imageData = ctx.createImageData(width, height);
+      imageData.data.set(pixels);
+      ctx.putImageData(imageData, 0, 0);
+      canvas.dataset.rendered = true;
+    },
+  };
+};
 
-let csrfToken = document.querySelector("meta[name='csrf-token']").getAttribute("content")
+let csrfToken = document
+  .querySelector("meta[name='csrf-token']")
+  .getAttribute("content");
 let liveSocket = new LiveSocket("/live", Socket, {
-    params: { _csrf_token: csrfToken },
-    dom: {
-        onBeforeElUpdated(from, to) {
-            if (from.__x) { Alpine.clone(from.__x, to) }
-        }
+  params: {
+    _csrf_token: csrfToken,
+  },
+  dom: {
+    onBeforeElUpdated(from, to) {
+      if (from.__x) {
+        Alpine.clone(from.__x, to);
+      }
+    },
+  },
+});
+
+window.nativeIOSWrapper = {
+  // The native code bridge assumes that handlers have been setup. Seethe docs for more info:
+  // https://developer.apple.com/documentation/webkit/wkusercontentcontroller/1537172-add
+  //
+  // Uncomment each section to enable it.
+  setScreenState: (state) => {
+    window.webkit.messageHandlers.Native.postMessage({
+      type: "setScreenState",
+      state,
```

Comment:
For next version it would be nice to have the screenId also available for setting state to prevent situations where state is set on the wrong screen during transition (pushing/popping)

Reply:
Done